source/_posts/2024-01-13-leaky-bucket.markdown

```diff
@@ -0,0 +1,108 @@
+---
+title: "Leaky Bucket Rate Limiting"
+kind: article
+created_at: 2024-01-13 19:40:00 UTC
+author: Shivam Singhal
+layout: post
+---
+Hmm…leaky bucket? The first time I heard these words, it made me chuckle. I thought what kind of name is this for a rate limiting strategy.
+
+But upon exploring it, it’s rather a very simple, elegant, and popular strategy used in a lot of production environments.
+
+But what are we rate limiting? It’s obvious that we want to rate limit the incoming requests. But do we rate limit the whole server(due to hardware constraints) or do we rate limit per user?
+
+The answer, well, depends on your business requirement. How do you differentiate between different users? An easy way out for us is that we can rate limit based on the IP Address of the request and that’s what we are going to do.
+
+<aside>
+💡 Bonus: check out an implementation of leaky bucket at the bottom!
```

Comment:
This sounds like it's trying to drive engagement a bit like how `watch till the end of this video` does 😅. It's alright, but try saying it in a way where you can avoid that connotation?

Reply:
True. Let me try to remove this connotation.